storm-core/src/jvm/org/apache/storm/topology/base/BaseWindowedBolt.java

```diff
@@ -66,10 +66,14 @@ public String toString() {
      * Holds a Time duration for time based windows and sliding intervals.
      */
     public static class Duration {
-        public final int value;
+        public final long value;
 
         public Duration(int value, TimeUnit timeUnit) {
-            this.value = (int) timeUnit.toMillis(value);
+            this.value = timeUnit.toMillis(value);
+        }
+
+        public Duration(long value) {
```

Comment:
Please add TimeUnit as a second arg instead of automatically assuming the time unit. Also have an overloaded `public static Duration of(long milliseconds)`. For other static factories long does not make much sense.